spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -835,6 +842,186 @@ public void setSchemaType(SchemaType schemaType) {
+			this.useKeyStoreTls = useKeyStoreTls;
+		}
+
+		public String getSslProvider() {
+			return this.sslProvider;
+		}
+
+		public void setSslProvider(String sslProvider) {
+			this.sslProvider = sslProvider;
+		}
+
+		public String getTlsTrustStoreType() {
+			return this.tlsTrustStoreType;
+		}
+
+		public void setTlsTrustStoreType(String tlsTrustStoreType) {
+			this.tlsTrustStoreType = tlsTrustStoreType;
+		}
+
+		public String getTlsTrustStorePath() {
+			return this.tlsTrustStorePath;
+		}
+
+		public void setTlsTrustStorePath(String tlsTrustStorePath) {
+			this.tlsTrustStorePath = tlsTrustStorePath;
+		}
+
+		public String getTlsTrustStorePassword() {
+			return this.tlsTrustStorePassword;
+		}
+
+		public void setTlsTrustStorePassword(String tlsTrustStorePassword) {
+			this.tlsTrustStorePassword = tlsTrustStorePassword;
+		}
+
+		public String[] getTlsCiphers() {
+			return this.tlsCiphers;
+		}
+
+		public void setTlsCiphers(String[] tlsCiphers) {
+			this.tlsCiphers = tlsCiphers;
+		}
+
+		public String[] getTlsProtocols() {
+			return this.tlsProtocols;
+		}
+
+		public void setTlsProtocols(String[] tlsProtocols) {
+			this.tlsProtocols = tlsProtocols;
+		}
+
+		public long getConnectionTimeoutMs() {
+			return this.connectionTimeoutMs;
+		}
+
+		public void setConnectionTimeoutMs(long connectionTimeoutMs) {
+			this.connectionTimeoutMs = connectionTimeoutMs;
+		}
+
+		public long getRequestTimeoutMs() {
+			return this.requestTimeoutMs;
+		}
+
+		public void setRequestTimeoutMs(long requestTimeoutMs) {
+			this.requestTimeoutMs = requestTimeoutMs;
+		}
+
+		public Map<String, Object> buildProperties() {
+			PulsarProperties.Properties properties = new Properties();
+
+			PropertyMapper map = PropertyMapper.get().alwaysApplyingWhenNonNull();
+
+			map.from(this::getServiceUrl).to(properties.in("serviceUrl"));
+			map.from(this::getAuthPluginClassName).to(properties.in("authPluginClassName"));
+			map.from(this::getAuthParams).to(properties.in("authParams"));
+			map.from(this::getTlsTrustCertsFilePath).to(properties.in("tlsTrustCertsFilePath"));
+			map.from(this::isTlsAllowInsecureConnection).to(properties.in("tlsAllowInsecureConnection"));
+			map.from(this::isTlsHostnameVerificationEnable).to(properties.in("tlsHostnameVerificationEnable"));
+			map.from(this::isUseKeyStoreTls).to(properties.in("useKeyStoreTls"));
+			map.from(this::getSslProvider).to(properties.in("sslProvider"));
+			map.from(this::getTlsTrustStoreType).to(properties.in("tlsTrustStoreType"));
+			map.from(this::getTlsTrustStorePath).to(properties.in("tlsTrustStorePath"));
+			map.from(this::getTlsTrustStorePassword).to(properties.in("tlsTrustStorePassword"));
+			map.from(this::getTlsCiphers).to(properties.in("tlsCiphers"));
+			map.from(this::getTlsProtocols).to(properties.in("tlsProtocols"));
+			map.from(this::getConnectionTimeoutMs).to(properties.in("connectionTimeoutMs"));
+			map.from(this::getRequestTimeoutMs).to(properties.in("requestTimeoutMs"));
```

Comment:
I notice that there are a handful of properties that are 1st class props on the `PulsarAdminImpl` that are **also** in the config data map. It seems that some of the ones in the map are not used and instead only the ones directly on the admin impl are used. 

```java
this.connectTimeout = connectTimeout;
this.connectTimeoutUnit = connectTimeoutUnit;
this.readTimeout = readTimeout;
this.readTimeoutUnit = readTimeoutUnit;
this.requestTimeout = requestTimeout;
this.requestTimeoutUnit = requestTimeoutUnit;
```

The 1st class ones are only set using the values from the AdminBuilder, not the map of props that we are using. The issue I see is that users may try to set the requestTimeout but then will be disappointed when it does not actually get used. 

On the regular `PulsarClient` side of things this is not an issue as the conf map seems to be the final arbiter of chosen values.

Reply:
Good catch! This is an oversight on our part, I'm opening a bug ticket for the builder.
For now I will remove the timeouts from the PulsarProperties to prevent this issue.